Added code for jump_search

ajogseneca opened the pull request:
## Jump Search
Jump search is an efficient algorithm designed for searching an element in a sorted array. It works by dividing the array into blocks of a fixed size (jump size), and then checks the elements in increments (or "jumps") of that size. Once it finds a block that contains the target, it performs a linear search within that block.

ajogseneca committed:
Added code for jump_search

FerdausPolok commented:
Hello! Is this PR ready for review? I see no description.

ajogseneca commented:
This PR is associated with Issue #34 

@FerdausPolok this PR is ready for review !

ajogseneca commented:
@FerdausPolok  Do we have a discord for this repo ?

ajogseneca commented:
@FerdausPolok Do we have any updates on this PR ?

FerdausPolok commented on algorithm/searching-algo-python/jump_search.py:
```diff
@@ -0,0 +1,22 @@
+import math
```
Do we have `math` on our dir?

FerdausPolok commented on algorithm/searching-algo-python/jump_search.py:
```diff
@@ -0,0 +1,22 @@
+
+def jump_search(arr, target):
+    n = len(arr)
+    step = int(math.sqrt(n))
```
Can we do this manually instead of using `math`?

FerdausPolok commented on algorithm/searching-algo-python/jump_search.py:
```diff
@@ -0,0 +1,22 @@
+    return -1
+
+# Example usage
+arr = [1,20,300,4000,5000]
```
Instead of hard coded array, can we take input from users?

FerdausPolok commented:
> @FerdausPolok Do we have any updates on this PR ?

requested some changes, sorry for the delay.

ajogseneca committed:
Updated with required changes.

ajogseneca commented:
## Changes
- I have added a function to calculate the square root manually instead of math lib
- I have added the feature to get input from the user and print respective messages back to the user.

ajogseneca commented:
@FerdausPolok take a look at the changes, Thankyou

ajogseneca commented:
@FerdausPolok can we merge this issue as soon as possible ?, Let me know if you want any more changes thank you !
